dfs/sudoku_solver.py

```diff
@@ -21,59 +21,95 @@
 for the board lol).
 """
 
-def solveSudoku(self, board):
-    self.board = board
-    self.val = self.PossibleVals()
-    self.Solver()
-
-def PossibleVals(self):
-    a = "123456789"
-    d, val = {}, {}
-    for i in xrange(9):
-        for j in xrange(9):
-            ele = self.board[i][j]
-            if ele != ".":
-                d[("r", i)] = d.get(("r", i), []) + [ele]
-                d[("c", j)] = d.get(("c", j), []) + [ele]
-                d[(i//3, j//3)] = d.get((i//3, j//3), []) + [ele]
-            else:
-                val[(i,j)] = []
-    for (i,j) in val.keys():
-        inval = d.get(("r",i),[])+d.get(("c",j),[])+d.get((i/3,j/3),[])
-        val[(i,j)] = [n for n in a if n not in inval ]
-    return val
-
-def Solver(self):
-    if len(self.val)==0:
+import unittest
+
+try:
+    xrange
+except:
+    xrange = range
```

Comment:
I don't think we should use `xrange` anymore. Please also change the usages of `xrange` to `range` for Python 3. Its also compatible with both versions 2 and 3.